### PR TITLE
azurestack-Network 2015-06-15 swagger getting split in to multiple files matching the latest

### PR DESCRIPTION
**Pull request by bganapa**
2015-06-15 swagger getting split in to multiple files matching the latest and also adding descriptions, creating a new composite network swagger

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

---
This change is [Reviewable](https://reviewable.io/reviews/azure/azure-rest-api-specs/1160)

### CODE REVIEW

**Comment by bganapa** [comment]
@jhendrixMSFT could you please help look at this error? I cant seem to find out the reason..

**Comment by bganapa** [comment]
@sergey-shandar could you please help look at this error? I cant seem to find out the reason..

**Review by sergey-shandar — COMMENTED** [review]
@bganapa as I can see, there are new files which using  the old api-version. Could you provide a little bit of context, why is that?

**Review by sergey-shandar — COMMENTED** [review]
@bganapa There are also build errors:

https://travis-ci.org/Azure/azure-rest-api-specs/jobs/225489481

```
1) External file or url references ("$ref") in a swagger spec: /home/travis/build/Azure/azure-rest-api-specs/arm-network/2015-06-15/swagger/virtualNetworkGateway.json should be completely resolvable.:
     Uncaught Error: /home/travis/build/Azure/azure-rest-api-specs/arm-network/2015-06-15/swagger/virtualNetworkGateway.json has references that cannot be resolved. They are as follows: 
'Error resolving $ref pointer "/home/travis/build/Azure/azure-rest-api-specs/arm-network/2015-06-15/swagger/virtualNetworkGateway.json#/definitions/Resource". \nToken "Resource" does not exist.'
      at test/syntax.js:142:17
      at node_modules/call-me-maybe/index.js:13:28
      at _combinedTickCallback (internal/process/next_tick.js:73:7)
      at process._tickCallback (internal/process/next_tick.js:104:9)
```

**Comment by bganapa** [comment]
@sergey-shandar Thanks, Could you please let me know how I could run this locally so that I can catch before I push?

**Comment by sergey-shandar** [comment]
@bganapa here's a link how to use the tool https://github.com/Azure/oav#how-to-run-the-tool

Come back to potential breaking changes. Because the changes are big, it's hard to prove that the changes does [not] contain breaking changes for SDKs. I would suggest to ship these kind of changes with new api-version.

**Comment by sergey-shandar** [comment]
@bganapa should we close the PR?

**Comment by bganapa** [comment]
@sergey-shandar , I am going to work on this PR tomorrow. I will sync up with you offline on why it is needed to keep the same api-version.

**Comment by sergey-shandar** [comment]
@bganapa would you still like to sync with me on this PR offline?

**Comment by bganapa** [comment]
@sergey-shandar  synced with sergey offline.  The changes are being to be consumed only by AzureStack. All the azure tools are pointing to the latest api-version swagger.  AzureStack supports only the old api-version 2015-06-15 . We have already generated Python SDK consuming these changes from my fork.  
https://github.com/Azure/azure-sdk-for-python/tree/master/azure-mgmt-network/azure/mgmt/network/v2015_06_15 ( Python SDK is multi versioned for the support of azure stack, rest like C# SDK is still single versioned and points only to the latest)
This Python SDK  is already being consumed in the CLI (master branch)

**Comment by sergey-shandar** [comment]
@amarzavery Do you think the Nodejs SDK will be ok with these changes?

File: arm-network/2015-06-15/swagger/virtualNetwork.json [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+        "dnsServers": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The list of DNS servers IP addresses."
+        }
+      },
+      "description": "DhcpOptions contains an array of DNS servers available to VMs deployed in the virtual network. Standard DHCP option for a subnet overrides VNET DHCP options."
+    },
```
Comment (sergey-shandar) [inline arm-network/2015-06-15/swagger/virtualNetwork.json]: The file is invalid JSON. I fixed it in an additional commit.

**Comment by bganapa** [comment]
cc:@amarzavery for the Q from sergey Do you think the Nodejs SDK will be ok with these changes?

**Comment by sergey-shandar** [comment]
@balajikris could you have a look the PR to make sure it will not cause problems for Nodejs SDK?

**Comment by balajikris** [comment]
@sergey-shandar -  I don't understand why we're editing a `2015` version when there are multiple newer versions, nor do I understand why we're keeping the version number as is. I feel I'm missing a lot of context on this change to make a call. 

FWIW, please note that any impact on the node sdk also impacts **azure/azure-xplat-cli** [tagging @EvgenyAgafonchikov for that]. 

In all, I think @amarzavery still the right person to make this call.

**Review by sergey-shandar — APPROVED** [review]
I've checked the diffs against the old network.json. Looks good to me.

**Comment by bganapa** [comment]
@sergey-shandar Thanks for the taking care of this

**Pull request merged**